### PR TITLE
Auto Completion for Prolog

### LINKED ISSUE #85: Auto Complete/ Completion for XML Prolog
When `<?` or `<?xml` is typed give the option to complete the rest

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #85

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java [hunk 1]
```diff
@@ -350,6 +355,28 @@ private void collectOpenTagSuggestions(boolean hasOpenBracket, Range replaceRang
...
+			item.setInsertTextFormat(InsertTextFormat.Snippet);
+			int closingBracketOffset = getOffsetFollowedBy(document.getText(), tagNameEnd, ScannerState.WithinTag, TokenType.StartTagClose);
+			int end = closingBracketOffset != -1 ? closingBracketOffset - 1 : tagNameEnd;
+			String closeTag = closingBracketOffset != -1 ? "" : ">";
+			try {
+				Range editRange = getReplaceRange(tagNameEnd, end, request);
+				item.setTextEdit(new TextEdit(editRange, " version=\"1.0\" encoding=\"UTF-8\"?" + closeTag + "$0"));
+			} catch (BadLocationException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
```
Comment (angelozerr) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java]: Use logger.

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java [hunk 2]
```diff
@@ -350,6 +355,28 @@ private void collectOpenTagSuggestions(boolean hasOpenBracket, Range replaceRang
 		}
 	}
 
+	private void createPrologCompletionSuggestion(int tagNameEnd, String tag, CompletionRequest request,
+		CompletionResponse response) {
```
Comment (angelozerr) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java]: Perhaps it's for an another PR, but IMHO I think we should have too completion on 

 * version attribute name
 * version attribute value (1.0, 2.0)
 * encoding  attribute name
 * encoding  attribute value (UTF-8, UTF-16, etc)

Comment (NikolasKomonen) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java]: I'll create an issue for that.

**Comment by angelozerr** [comment]
Thanks @NikolasKomonen !

**Pull request merged**